src/org/zaproxy/zap/extension/groovy/ExtensionGroovy.java

```diff
@@ -0,0 +1,132 @@
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptNode;
+import org.zaproxy.zap.extension.script.ScriptType;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+
+import javax.script.ScriptEngine;
+import javax.swing.ImageIcon;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ExtensionGroovy extends ExtensionAdaptor {
+
+	public static final String NAME = "ExtensionGroovy";
+	public static final int EXTENSION_ORDER = 83;
+	public static final ImageIcon GROOVY_ICON;
+
+	private static final Logger LOGGER = Logger.getLogger(ExtensionGroovy.class);
+	private static final List<Class<? extends Extension>> EXTENSION_DEPENDENCIES;
+
+	static {
+		List<Class<? extends Extension>> dependencies = new ArrayList<>(1);
+		dependencies.add(ExtensionScript.class);
+		EXTENSION_DEPENDENCIES = Collections.unmodifiableList(dependencies);
+
+		GROOVY_ICON = View.isInitialised()
+				? new ImageIcon(ExtensionGroovy.class.getResource("/org/zaproxy/zap/extension/groovy/resources/groovy.png"))
+				: null;
+	}
+
+	private ExtensionScript extScript = null;
+	private GroovyEngineWrapper groovyEngine = null;
+
+	public ExtensionGroovy() {
+		super(NAME);
+		this.setOrder(EXTENSION_ORDER);
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+		super.hook(extensionHook);
+		ScriptEngine se = getGroovyScriptEngine();
+		if (se != null) {
+			groovyEngine = new GroovyEngineWrapper(se);
+			getExtScript().registerScriptEngineWrapper(groovyEngine);
+		} else {
+			LOGGER.error("No Groovy engine found");
+		}
+	}
+
+	private ScriptEngine getGroovyScriptEngine()
+	{
+		return GroovyEngineWrapper.getScriptEngineFromFactory();
+	}
+
+	@Override
+	public boolean canUnload() {
+		return true;
+	}
+
+	@Override
+	public void unload() {
+		super.unload();
+		if(groovyEngine != null){
+			getExtScript().removeScriptEngineWrapper(groovyEngine);
+		}
+	}
+
+	private ExtensionScript getExtScript() {
+		if (extScript == null) {
+			extScript = (ExtensionScript) Control
+				.getSingleton()
+				.getExtensionLoader()
+				.getExtension(ExtensionScript.NAME);
+		}
+		return extScript;
+	}
+
+	@Override
+	public String getAuthor() {
+		return Constant.ZAP_TEAM;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("groovy.desc");
+	}
+
+	@Override
+	public URL getURL() {
+		try {
+			return new URL(Constant.ZAP_HOMEPAGE);
+		} catch (MalformedURLException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public List<Class<? extends Extension>> getDependencies() {
+		return EXTENSION_DEPENDENCIES;
+	}
+}

```

Comment:
Should also include an overridden `getUIName()` that returns a user friendly extension name for use in Options > Extensions (and I'm not sure where else ;) )